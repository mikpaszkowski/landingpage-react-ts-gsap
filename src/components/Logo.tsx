import * as React from "react";
import { ReactComponent as Ellipse2 } from "../assets/images/svg-images/ellipse-2.svg";
import { ReactComponent as Ellipse1 } from "../assets/images/svg-images/ellipse-1.svg";
import styled from "styled-components";
import { colors } from "../styles/styleConstants";

const LogoWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  position: relative;
  width: 17.3rem;
  height: 4.7rem;
`;

const LogoText = styled.p`
  font-size: 4rem;
  font-style: normal;
  font-weight: normal;
  text-align: center;
  width: 10.2rem;
  line-height: 117%;
  color: ${(props: LogoProps) =>
    props.inverse ? "white" : colors.primaryFont};
`;

const Ellipse1Absolute = styled(Ellipse1)`
  position: absolute;
  top: 0.7rem;
  right: 0.3rem;
`;

interface LogoProps {
  inverse?: boolean | null;
}

export const Logo = ({ inverse }: LogoProps) => {
  return (
    <LogoWrapper>
      <Ellipse1Absolute />
      <Ellipse2 />
      <LogoText inverse={inverse}>Medli</LogoText>
    </LogoWrapper>
  );
};
