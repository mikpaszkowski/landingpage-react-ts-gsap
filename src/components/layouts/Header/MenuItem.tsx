import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../../styles/styleConstants";
import { ReactComponent as FocusIcon } from "./ellipse-4.svg";

const MenuItemWrapper = styled.div`
  margin: 0.7rem 0;
`;

const MenuItemText = styled.a`
  position: relative;
  font-family: ${fontTypes.secondaryFont};
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
  text-decoration: none;
  color: ${({ theme }) => theme.palette.primary.font};

  &:focus > svg {
    display: block;
  }
`;

const FocusSVG = styled(FocusIcon)`
  display: none;
  position: absolute;
  bottom: -0.8rem;
  left: 50%;
  transform: translateX(-50%);
`;

interface MenuItemProps {
  name: string;
}

export const MenuItem = ({ name }: MenuItemProps) => {
  return (
    <MenuItemWrapper>
      <MenuItemText href="#">
        {name}
        <FocusSVG />
      </MenuItemText>
    </MenuItemWrapper>
  );
};
