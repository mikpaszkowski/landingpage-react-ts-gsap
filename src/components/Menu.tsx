import * as React from "react";
import styled from "styled-components";
import { MenuItem } from "./MenuItem";
import { CustomButton } from "./CustomButton";

const MenuWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 64.4rem;
`;

export const Menu = () => {
  return (
    <MenuWrapper>
      <MenuItem name="Home" />
      <MenuItem name="About" />
      <MenuItem name="Courses" />
      <MenuItem name="News" />
      <MenuItem name="Contact" />
      <CustomButton fontsize="2.4rem" lineHeight="3.6rem">
        Log in
      </CustomButton>
    </MenuWrapper>
  );
};
