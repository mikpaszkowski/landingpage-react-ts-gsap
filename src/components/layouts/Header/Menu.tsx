import * as React from "react";
import styled from "styled-components";
import { MenuItem } from "./MenuItem";
import { CustomButton } from "../../ReusableComponents/CustomButton";
import { device } from "../../../styles/responsive";

const MenuWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 64.4rem;

  @media ${device.mobileMedium} {
    display: none;
  }
`;

export const Menu = () => {
  return (
    <MenuWrapper>
      <MenuItem name="Home" />
      <MenuItem name="About" />
      <MenuItem name="Courses" />
      <MenuItem name="News" />
      <MenuItem name="Contact" />
      <CustomButton fontsize="2.4rem" lineHeight="3.6rem">
        Log in
      </CustomButton>
    </MenuWrapper>
  );
};
