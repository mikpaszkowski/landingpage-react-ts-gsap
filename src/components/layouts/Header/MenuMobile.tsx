import * as React from "react";
import styled, { keyframes } from "styled-components";
import { MenuItem } from "./MenuItem";
import { CustomButton } from "../../ReusableComponents/CustomButton";
import { menuItems, IMenuItems } from "../../../assets/menuItems/menuItems";
import { IoCloseOutline } from "react-icons/io5";

const slideDown = keyframes`
  from{
    transform: translateY(-100%);
  }
  to{
    transform: translateY(0);
  }
`;

const MenuWrapper = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  z-index: 999;
  padding: 7rem 2.4rem 5rem 2.4rem;
  background-color: ${({ theme }) => theme.palette.common.background};
  transform: translateY(0);
  animation-name: ${slideDown};
  animation-duration: 0.4s;
  animation-timing-function: cubic-bezier(0.445, 0.05, 0.55, 0.95);
  animation-fill-mode: both;
  transition: all 1s ease-in-out;
`;

const Menu = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  height: 100%;
`;

const LogInButton = styled(CustomButton)`
  margin-top: 4rem;
  width: 100%;
`;

const CloseButton = styled(IoCloseOutline)`
  position: absolute;
  top: 2.7rem;
  right: 2.4rem;
  width: 5rem;
  height: 5rem;
  cursor: pointer;
  transition: transform 0.3s ease-in-out;

  &:hover {
    transform: rotate(90deg);
  }
`;

interface MenuMobileProps {
  isMenuOpen: boolean | null;
  closeMenuMobile: () => void;
  menuItems: Array<IMenuItems>;
}

export const MenuMobile = ({
  isMenuOpen,
  closeMenuMobile,
  menuItems,
}: MenuMobileProps) => {
  return isMenuOpen ? (
    <MenuWrapper>
      <Menu>
        {menuItems.forEach(({ id, name }) => {
          <MenuItem key={id} name={name} />;
        })}
        <LogInButton fontSize="2.4rem" lineHeight="3.6rem">
          Log in
        </LogInButton>
      </Menu>
      <CloseButton onClick={closeMenuMobile} />
    </MenuWrapper>
  ) : null;
};
