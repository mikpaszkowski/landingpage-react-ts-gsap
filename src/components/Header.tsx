import * as React from "react";
import { Logo } from "./Logo";
import { Menu } from "./Menu";
import styled from "styled-components";
import { ReactComponent as Bar } from "../assets/images/svg-images/menu-bar.svg";
import { device } from "../styles/responsive";
import { MenuMobile } from "./MenuMobile";
import { colors } from "../styles/styleConstants";

const HeaderWrapper = styled.header`
  position: fixed;
  top: 0;
  left: 50%;
  z-index: 10;
  max-width: 145rem;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 0 auto;
  padding: 2.7rem 16.7rem;
  transform: translateX(-50%);
  background-color: ${colors.background};

  @media ${device.mobileMedium} {
    position: relative;
    padding: 2.7rem 2.4rem;
  }
`;
const MenuBar = styled(Bar)`
  display: none;
  width: 3rem;
  height: 1.7rem;
  cursor: pointer;

  @media ${device.mobileMedium} {
    display: block;
    position: absolute;
    top: 50%;
    right: 2.4rem;
    transform: translateY(-50%);
  }
`;

const { useState } = React;

export const Header = () => {
  const [menuBarOpen, setMenuBarOpen] = useState<boolean | null>(null);

  const handleClick = () => {
    setMenuBarOpen(!menuBarOpen);
  };

  return (
    <HeaderWrapper>
      <Logo />
      <Menu />
      <MenuBar onClick={handleClick} />
      <MenuMobile isMenuOpen={menuBarOpen} closeMenuMobile={handleClick} />
    </HeaderWrapper>
  );
};
