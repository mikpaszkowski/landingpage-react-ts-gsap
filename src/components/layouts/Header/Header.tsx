import * as React from "react";
import { Logo } from "../../ReusableComponents/Logo/Logo";
import { Menu } from "./Menu";
import styled from "styled-components";
import { ReactComponent as Bar } from "./menu-bar.svg";
import { menuItems } from "../../staticContents/menu/menuItems";
import { device } from "../../../styles/responsive";
import { MenuMobile } from "./MenuMobile";

const HeaderWrapper = styled.header`
  position: fixed;
  top: 0;
  left: 50%;
  z-index: 9999;
  max-width: 145rem;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 0 auto;
  padding: 2.7rem 16.7rem;
  transform: translateX(-50%);
  background-color: ${({ theme }) => theme.palette.common.background};

  @media ${device.laptopSmall} {
    padding: 2.7rem 6rem;
  }

  @media ${device.mobileMedium} {
    position: relative;
    padding: 2.7rem 2.4rem;
  }
`;
const MenuBar = styled(Bar)`
  display: none;
  width: 3rem;
  height: 1.7rem;
  cursor: pointer;
  fill: ${({ theme }) => theme.palette.primary.font};

  @media ${device.tabletLarge} {
    display: block;
    position: absolute;
    top: 50%;
    right: 2.4rem;
    transform: translateY(-50%);
    fill: ${({ theme }) => theme.palette.primary.font};
  }
`;

const { useState } = React;

export const Header = () => {
  const [menuBarOpen, setMenuBarOpen] = useState<boolean | null>(null);

  const handleClick = () => {
    setMenuBarOpen(!menuBarOpen);
  };

  return (
    <HeaderWrapper>
      <Logo />
      <Menu menuItems={menuItems} />
      <MenuBar onClick={handleClick} />
      <MenuMobile
        isMenuOpen={menuBarOpen}
        closeMenuMobile={handleClick}
        menuItems={menuItems}
      />
    </HeaderWrapper>
  );
};
