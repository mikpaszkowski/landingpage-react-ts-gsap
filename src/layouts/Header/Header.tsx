import * as React from "react";
import { Logo } from "../../components/Logo/Logo";
import { Menu } from "./Menu";
import styled from "styled-components";
import { useTheme } from "../../hooks/useTheme";
import { lightTheme } from "../../styles/theme";
import { ReactComponent as BlackBar } from "../../assets/svg/menu-bar-black.svg";
import { ReactComponent as WhiteBar } from "../../assets/svg/menu-bar-black.svg";
import { menuItems } from "../../data/staticContents/menu/menuItems";
import { device } from "../../styles/responsive";
import { MenuMobile } from "./MenuMobile";

const HeaderWrapper = styled.header`
  position: fixed;
  top: 0;
  left: 50%;
  z-index: 9999;
  max-width: 145rem;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 0 auto;
  padding: 2.7rem 16.7rem;
  transform: translateX(-50%);
  background-color: ${({ theme }) => theme.palette.common.background};

  @media ${device.laptopSmall} {
    padding: 2.7rem 6rem;
  }

  @media ${device.mobileMedium} {
    position: relative;
    padding: 2.7rem 2.4rem;
  }
`;
const BlackMenuBar = styled(BlackBar)`
  display: none;
  width: 3rem;
  height: 1.7rem;
  cursor: pointer;

  @media ${device.tabletLarge} {
    display: block;
    position: absolute;
    top: 50%;
    right: 2.4rem;
    transform: translateY(-50%);
  }
`;

const WhiteMenuBar = styled(WhiteBar)`
  display: none;
  width: 3rem;
  height: 1.7rem;
  cursor: pointer;
  color: white;

  svg {
    fill: white;
  }

  @media ${device.tabletLarge} {
    display: block;
    position: absolute;
    top: 50%;
    right: 2.4rem;
    transform: translateY(-50%);
    fill: ${({ theme }) => theme.palette.primary.font};
  }
`;

const { useState } = React;

export const Header = () => {
  const [menuBarOpen, setMenuBarOpen] = useState<boolean | null>(null);
  const { theme } = useTheme();

  const handleClick = () => {
    setMenuBarOpen(!menuBarOpen);
  };

  return (
    <HeaderWrapper>
      <Logo />
      <Menu menuItems={menuItems} />
      {theme === lightTheme ? (
        <BlackMenuBar onClick={handleClick} />
      ) : (
        <WhiteMenuBar onClick={handleClick} />
      )}
      <MenuMobile
        isMenuOpen={menuBarOpen}
        closeMenuMobile={handleClick}
        menuItems={menuItems}
      />
    </HeaderWrapper>
  );
};
