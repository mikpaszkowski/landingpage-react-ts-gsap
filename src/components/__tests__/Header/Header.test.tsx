import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import { renderWithLightTheme, mountWithLightTheme } from "../helpers/index";
import { Header } from "../../../layouts/Header/Header";
import { Logo } from "../../Logo/Logo";
import { Menu } from "../../../layouts/Header/Menu";
import { MenuMobile } from "../../../layouts/Header/MenuMobile";

afterEach(() => {
  cleanup();
});

test("check render correctly <Header /> component", () => {
  const tree = renderWithLightTheme(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("check Header component", () => {
  const wrapper = mountWithLightTheme(<Header />);

  test("should contain three div elements / component", () => {
    expect(wrapper.find("div").at(0).children().length).toEqual(3);
  });
  test("should contain one svg component - Logo", () => {
    expect(wrapper.find("svg").at(0).children().length).toEqual(1);
  });

  test("should contain one Logo component", () => {
    expect(wrapper.find(Logo).length).toEqual(1);
  });

  test("should contain one Menu component", () => {
    expect(wrapper.find(Menu).length).toEqual(1);
  });

  test("should contain one MenuMobile component", () => {
    expect(wrapper.find(MenuMobile).length).toEqual(1);
  });
});
