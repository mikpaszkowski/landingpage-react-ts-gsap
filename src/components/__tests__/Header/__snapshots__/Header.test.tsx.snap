// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render correctly <Header /> component 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  width: 17.3rem;
  height: 4.7rem;
  cursor: pointer;
}

.c3 {
  font-size: 4rem;
  font-style: normal;
  font-weight: normal;
  text-align: center;
  width: 10.2rem;
  line-height: 117%;
  color: #131E07;
}

.c2 {
  position: absolute;
  top: 0.7rem;
  right: 0.3rem;
}

.c5 {
  margin: 0.7rem 0;
}

.c6 {
  position: relative;
  font-family: 'Reem Kufi',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #131E07;
}

.c6:focus > svg {
  display: block;
}

.c7 {
  display: none;
  position: absolute;
  bottom: -0.8rem;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c8 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.7rem 3.1rem;
  font-family: 'Reem Kufi',sans-serif;
  font-size: 2.4rem;
  font-style: normal;
  font-weight: normal;
  line-height: 3.6rem;
  color: #131E07;
  border: none;
  border-radius: 8px;
  background-color: #FABD20;
  cursor: pointer;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  overflow: hidden;
  will-change: transform;
}

.c8::after {
  content: "";
  position: absolute;
  z-index: -1;
  bottom: -100%;
  left: 0;
  height: 100%;
  width: 100%;
  border-radius: 8px;
  background-color: black;
  color: #FABD20;
  -webkit-transition: all 0.15s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: all 0.15s cubic-bezier(0.455,0.03,0.515,0.955);
}

.c8:hover {
  box-shadow: 0px 1rem 2rem rgba(0,0,0,0.25);
  color: #FABD20;
}

.c8:hover::after {
  -webkit-transform: translateY(-100%);
  -ms-transform: translateY(-100%);
  transform: translateY(-100%);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 64.4rem;
}

.c0 {
  position: fixed;
  top: 0;
  left: 50%;
  z-index: 9999;
  max-width: 145rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 auto;
  padding: 2.7rem 16.7rem;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  background-color: #FEF4E6;
}

.c9 {
  display: none;
  width: 3rem;
  height: 1.7rem;
  cursor: pointer;
}

@media (max-width:1110px) {
  .c4 {
    display: none;
  }
}

@media (max-width:1280px) {
  .c0 {
    padding: 2.7rem 6rem;
  }
}

@media (max-width:375px) {
  .c0 {
    position: relative;
    padding: 2.7rem 2.4rem;
  }
}

@media (max-width:1110px) {
  .c9 {
    display: block;
    position: absolute;
    top: 50%;
    right: 2.4rem;
    -webkit-transform: translateY(-50%);
    -ms-transform: translateY(-50%);
    transform: translateY(-50%);
  }
}

<header
  className="c0"
>
  <div
    className="c1"
  >
    <svg
      className="c2"
    >
      ellipse-1.svg
    </svg>
    <svg>
      ellipse-2.svg
    </svg>
    <p
      className="c3"
    >
      Medli
    </p>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      <a
        className="c6"
        href="#"
      >
        Home
        <svg
          className="c7"
        >
          ellipse-4.svg
        </svg>
      </a>
    </div>
    <div
      className="c5"
    >
      <a
        className="c6"
        href="#"
      >
        About
        <svg
          className="c7"
        >
          ellipse-4.svg
        </svg>
      </a>
    </div>
    <div
      className="c5"
    >
      <a
        className="c6"
        href="#"
      >
        Courses
        <svg
          className="c7"
        >
          ellipse-4.svg
        </svg>
      </a>
    </div>
    <div
      className="c5"
    >
      <a
        className="c6"
        href="#"
      >
        News
        <svg
          className="c7"
        >
          ellipse-4.svg
        </svg>
      </a>
    </div>
    <div
      className="c5"
    >
      <a
        className="c6"
        href="#"
      >
        Contact
        <svg
          className="c7"
        >
          ellipse-4.svg
        </svg>
      </a>
    </div>
    <button
      className="c8"
      fontSize="2.4rem"
    >
      Log in
    </button>
  </div>
  <svg
    className="c9"
    onClick={[Function]}
  >
    menu-bar-black.svg
  </svg>
</header>
`;
